File: lib/ts/src/internal/validation/GetTypeUnexpectedValidationFailure.ts
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

import { ValidationFailure } from '../../internal/validation/ValidationFailure.js';
import { getType } from '../../internal/types/GetType.js';

export function getTypeUnexpectedValidationFailure(path: any[], value: any, expectedType: string): ValidationFailure[] {
    const actualType = getType(value);
    const data: { [key: string]: any } = {
        actual: { [actualType]: {} },
        expected: { [expectedType]: {} },
    };
    return [new ValidationFailure(path, 'TypeUnexpected', data)];
}

File: lib/py/telepact/internal/validation/GetTypeUnexpectedValidationFailure.py
#|
#|  Copyright The Telepact Authors
#|  SPDX-License-Identifier: Apache-2.0
#|

from ...internal.validation.ValidationFailure import ValidationFailure
from ...internal.types.GetType import get_type


def get_type_unexpected_validation_failure(path: list[object], value: object, expected_type: str) -> list['ValidationFailure']:
    actual_type = get_type(value)
    data: dict[str, object] = {
        "actual": {actual_type: {}},
        "expected": {expected_type: {}}
    }
    return [
        ValidationFailure(path, "TypeUnexpected", data)
    ]
